### ISSUE TITLE
Lack of bar, slider and buttons for demo playback !

### ISSUE THREAD

**Issue by garry-ut99** [stijn-volckaert/udemo#32]
This must be a good joke. This antediluvian fossil doesn't even has bar, slider and buttons for playback !
Currently it requires to type f..king commands in console ! Like in MS-DOS before Win 1.0 ! But we are in 2023, aren't we ?!.. I had a Sony walkman 25 years ago and it had buttons for playback ! Even my grandma's 30 yo TV had buttons ! For this reason, nobody use this udemo shitty thing and most of people use screen recording software to record and playback their gameplay, not even saying this udemo thing is broken due to other issues as well, the whole udemo thing is useless in it's current state.

> --- Description ---:  Unreal Tournament has a truly remarkable and complex demo recording and playback system.  There was only one major problem: WHERE IS THE BROWSER????

Yeah, also: HOUSTON, WE HAVE A PROBLEM: WHERE ARE THE F..KING: BAR, SLIDER AND BUTTONS FOR PLAYBACK???? WE CANNOT EASILY STEER OUR SHIP !

![](https://i.imgur.com/xWmiDAm.png)

![](https://i.imgur.com/9kfYHEw.png)

**Comment by prom6** [comment]
come on man, its a community project where some guys (1 at max 2) support a mod/mutator, because the original developer isnt supporting it anymore. the original demo playback feature that was built in UT, didnt even have the functions this mod has and there the developers got paid for developing
for example not even UT4 has something like this
saying its useless cause youre too lazy to keybind important commands in the user.ini is wild
the built in demo functionality on its own is already great (the recorded stuff is scalable and the demo files are under 30mb in size, no screen recording can do this, you can still obs record from demos though..) and this mod adds important stuff to it

this request is like the most unimportant thing i've read in this git, not to mention the way you 'demand' this feature is crazy disrespectful. if i was one of the developers i would either ignore it or delete it

if you cant live without a playback bar so badly then learn uscript, code it yourself and contribute to the project

**Comment by SeriousBuggie** [comment]
This can be done as floating panel, showed by some hotkey, like console do. Other variants not really good. Implement this is hud can be visible during record or overlap something.

**Comment by garry-ut99** [comment]
> prom6 

Sorry, I didn't mean to sound rude and I didn't say Demo feature is bad, only wanted to be able to playback in some human way.

> prom6 : too lazy to keybind important commands in the user.ini 
> prom6 : this request is like the most unimportant thing i've read in this git,

No, because your fu...ing bind in user.ini doesn't work ! Nor in xconsole.ini !
`A=slomo 0.5 ` or
`Aliases[28]=(Command="slomo 0.5",Alias="slomo") `
`A=slomo` or
`CustomButtons[3]=(Name="slomo 0.5",Command="slomo 0.5")`

**Comment by SeriousBuggie** [comment]
For me bind `slomo 0.5` on GreyPlus work in demoplay.

**Comment by garry-ut99** [comment]
Unfortunately, still doesn't work for me, neither A nor GreyPlus :(
even slomo 0.5 typed in xconsole while demoplay, doesn't work anymore, I remember it did work before...
also in some demos when pressing PAUSE button, it says on the chat "the game is unpausable"
and when clicking mouse "to switch view to other player" it doesn't work
I use client 469c on Win + udemo 3.5.1 + xconsole 3.4.8

**Comment by prom6** [comment]
For me the binds also work as intended:

2=slomo 0.5
3=slomo 1 | curtime
4=slomo 3
5=slomo 10

didnt bind seekto cause normally you want a specific number after you fastforwarded the demo and stopped at a specific time to return to this
I dont have any console mods active
Also the switch player and pause work as intended

And I have/had like a lot versions running from demomanager 2 to 3.5.1 and ut436 to 469c (screenshot :o )
never had any problems concerning the keybinds
so also clueless what goes wrong on your end

never fiddled with the aliases and stuff, but the normal keybind (key=slomo 0.5) should work, maybe you overwrote some basic command with the aliases? also if you just used the aliases one at a time, creating alias 'slomo' may interfere with the actual 'slomo 0.5' command, so this looks wrong to me (SeriousBuggie may know more on that), same goes for the custom button and then also 'slomo' for the alias may not work

do you use timebased, or framebased or fast as possible for demoplayback? maybe there is some weird stuff going on when having too high of a framerate


![image](https://github.com/stijn-volckaert/udemo/assets/71797130/9bea00cc-1c54-4298-8809-9a886a9caa1a)

**Comment by SeriousBuggie** [comment]
Need for start ensure udemo used for play demo, not stock demoplay driver.

**Comment by garry-ut99** [comment]
Indeed, forgot to switch from stock demo to udemo, but it's because I was forced to constant switching between stock demo and udemo due to various issues with udemo. Binding works now, but now I can't even watch a f.....g demo, because of this: 
- https://github.com/stijn-volckaert/udemo/issues/31

**Comment by garry-ut99** [comment]
Ok, given a fact key bindings via user.ini works fine, the feature request is not super important (althought would be nice), consider it a secondary priority.

**Comment by SeriousBuggie** [comment]
Idea for record: There can be some command during game play, which store marks in special file near demo.

Later this marks used for show on seek bar like chapters in video players, so user can fast seek to it.

**Comment by SeriousBuggie** [comment]
DemoPanel command open this floating window now.

<img width="485" height="239" alt="Image" src="https://github.com/user-attachments/assets/50ad7761-e8d0-48ed-a1bf-0274040a5ca5" />
